packages/authorization-platformstate-writer/src/consumerServiceV1.ts

```diff
@@ -1,20 +1,512 @@
+              };
+
+            await upsertTokenStateClientPurposeEntry(
+              newTokenClientPurposeEntry,
+              dynamoDBClient
+            );
+            await deleteClientEntryFromTokenGenerationStatesTable(
+              entry,
+              dynamoDBClient
+            );
+          } else if (entry.PK.startsWith(clientKidPurposePrefix)) {
+            const kid = extractKidFromTokenEntryPK(entry.PK);
+            if (!kidSet.has(kid)) {
+              const pk = makeTokenGenerationStatesClientKidPurposePK({
+                clientId,
+                kid,
+                purposeId,
+              });
+
+              const newClientPurposeEntry: TokenGenerationStatesClientPurposeEntry =
+                {
+                  consumerId: entry.consumerId,
+                  updatedAt: new Date().toISOString(),
+                  PK: pk,
+                  clientKind: entry.clientKind,
+                  publicKey: entry.publicKey,
+                  GSIPK_clientId: entry.GSIPK_clientId,
+                  GSIPK_kid: entry.GSIPK_kid,
+                  GSIPK_clientId_purposeId: makeGSIPKClientIdPurposeId({
+                    clientId,
+                    purposeId,
+                  }),
+                  GSIPK_purposeId: purposeId,
+                  purposeState: purposeEntry.state,
+                  purposeVersionId: purposeEntry.purposeVersionId,
+                  GSIPK_consumerId_eserviceId:
+                    agreementEntry.GSIPK_consumerId_eserviceId,
+                  agreementId: extractAgreementIdFromAgreementPK(
+                    agreementEntry.PK
+                  ),
+                  agreementState: agreementEntry.state,
+                  GSIPK_eserviceId_descriptorId:
+                    makeGSIPKEServiceIdDescriptorId({
+                      eserviceId: purposeEntry.purposeEserviceId,
+                      descriptorId: agreementEntry.agreementDescriptorId,
+                    }),
+                  descriptorState: catalogEntry.state,
+                  descriptorAudience: catalogEntry.descriptorAudience,
+                  descriptorVoucherLifespan:
+                    catalogEntry.descriptorVoucherLifespan,
+                };
+
+              await upsertTokenStateClientPurposeEntry(
+                newClientPurposeEntry,
+                dynamoDBClient
+              );
+
+              kidSet.add(kid);
+              // eslint-disable-next-line functional/immutable-data
+              addedTokenClientPurposeEntries.push(newClientPurposeEntry);
+            }
+          } else {
+            throw genericInternalError(`Unable to parse ${entry}`);
+          }
+
+          const secondRetrievalStates = await retrievePlatformStatesByPurpose(
+            purposeId,
+            dynamoDBClient
+          );
+
+          for (const clientPurposeEntry of addedTokenClientPurposeEntries) {
+            if (
+              secondRetrievalStates.agreementEntry.state !==
```

Comment:
[same](https://github.com/pagopa/interop-be-monorepo/pull/1062#discussion_r1819107878)

Reply:
Updated